src/skan/csr.py

```diff
@@ -1221,3 +1241,297 @@ def sholl_analysis(skeleton, center=None, shells=None):
     intersection_counts = np.bincount(shells, minlength=len(shell_radii)) // 2
 
     return center, shell_radii, intersection_counts
+
+
+def skeleton_to_nx(skeleton: Skeleton, summary: pd.DataFrame | None = None):
+    """Convert a Skeleton object to a networkx Graph."""
+    if summary is None:
+        summary = summarize(skeleton, separator='_')
+    # Ensure underscores in column names
+    summary.rename(columns=lambda s: s.replace('-', '_'), inplace=True)
+    g = nx.MultiGraph()
+    for row in summary.itertuples(name='Edge'):
+        i = (row.node_id_src)
+        j = (row.node_id_dst)
+        # Nodes are added if they don't exist so only need to add edges
+        g.add_edge(i, j, **{'path': skeleton.path_coordinates(row.Index)})
+    return g
+
+
+def nx_to_skeleton(g: nx.Graph | nx.MultiGraph, orig_dim: tuple) -> Skeleton:
+    """Convert a Networkx Graph to a Skeleton object.
+
+    The NetworkX Graph should have been created using skeleton_to_nx() function
+    which stores the co-ordinates of points between nodes as an edge property.
+    These are extracted and used to re-create the original Numpy array
+
+    Currently this is limited to 2D skeletons.
+
+
+    Parameters
+    ----------
+    g: nx.Graph
+        Networkx graph to convert to Skeleton.
+    orig_dim: tuple
+        Tuple of the original images dimensions
+
+    Returns
+    -------
+    Skeleton
+        Skeleton object.
+    """
+    np_img = np.zeros(orig_dim)
+    x, y = [], []
+    try:
+        for edge in g.edges.data(data='path', default=1):
```

Comment:
Why `default=1` here? There shouldn't be any edges without data in the skeleton? If there is one, then it's not a valid input skeleton (?) and we should probably just raise an error? (And this is equally easy to do with None as with 1?)

Reply:
Honestly can't remember now, sorry.

Have removed it and left it as default, tests pass ok.